src/utils/rpcQueries/viewFunction.ts

```diff
@@ -0,0 +1,51 @@
+import { providers } from 'near-api-js';
+import { printTxOutcomeLogs } from '@near-js/utils';
+
+// types
+import type { IOptions } from './types';
+
+type ViewFunctionResult =
+  | string
+  | number
+  | boolean
+  | null
+  | object
+  | unknown[]
+  | undefined;
+
+function parseJsonFromRawResponse(response: Uint8Array): ViewFunctionResult {
+  return JSON.parse(Buffer.from(response).toString());
+}
+
+export default async function viewFunction({
+  contractId,
+  method,
+  rpcURL,
+  args = {},
+}: IOptions): Promise<ViewFunctionResult> {
+  const url = rpcURL || `https://rpc.mainnet.near.org`;
+  const provider = new providers.JsonRpcProvider({ url });
+
+  const res = await provider.query({
+    request_type: 'call_function',
+    account_id: contractId,
+    method_name: method,
+    args_base64: Buffer.from(JSON.stringify(args)).toString('base64'),
```

Comment:
Like above, this will fail without a polyfill on browsers. I recommed we use a library to handle base64: https://www.npmjs.com/package/@stablelib/base64